cmd/tailscale/cli/dns-query.go

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+package cli
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"net/netip"
+
+	"golang.org/x/net/dns/dnsmessage"
+	"tailscale.com/types/dnstype"
+)
+
+func runDNSQuery(ctx context.Context, args []string) error {
+	if len(args) < 1 {
+		return flag.ErrHelp
+	}
+	name := args[0]
+	queryType := "A"
+	if len(args) >= 2 {
+		queryType = args[1]
+	}
+	fmt.Printf("DNS query for %q (%s) using internal resolver:\n", name, queryType)
+	bytes, resolvers, err := localClient.QueryDNS(ctx, name, queryType)
+	if err != nil {
+		fmt.Printf("failed to query DNS: %v\n", err)
+		return err
+	}
+	fmt.Println()
+	if len(resolvers) == 1 {
+		fmt.Printf("Forwarding to resolver: %v\n", printResolver(*resolvers[0]))
+	} else {
+		fmt.Println("Forwarding to multiple resolvers (race):")
+		for _, r := range resolvers {
+			fmt.Printf("  - %v\n", printResolver(*r))
+		}
+	}
+	fmt.Println()
+	var p dnsmessage.Parser
+	header, err := p.Start(bytes)
+	if err != nil {
+		fmt.Printf("failed to parse DNS response: %v\n", err)
+		return err
+	}
+	fmt.Printf("Response code: %v\n", header.RCode.String())
+	fmt.Println()
+	p.SkipAllQuestions()
+	if header.RCode != dnsmessage.RCodeSuccess {
+		fmt.Println("No answers were returned.")
+		return nil
+	}
+	answers, err := p.AllAnswers()
+	if err != nil {
+		fmt.Printf("failed to parse DNS answers: %v\n", err)
+		return err
+	}
+	if len(answers) == 0 {
+		fmt.Println("  (no answers found)")
+	}
+	for _, a := range answers {
+		fmt.Printf("%-40s %-5d %-10s %-10s %s\n", a.Header.Name.String(), a.Header.TTL, a.Header.Class.String(), a.Header.Type.String(), printAnswerBody(a))
+	}
```

Comment:
Up to you, but if you wanted, you could use [`text/tabwriter`](https://pkg.go.dev/text/tabwriter) to write this; something like: https://go.dev/play/p/I2HcdaOSYKS

Outputs (example):
````
Name        TTL  Class  Type  Body
----        ---  -----  ----  ----
google.com  300  INET   A     1.2.3.4
google.com  300  INET   A     8.8.8.8
google.com  300  INET   A     255.255.255.255
google.com  300  INET   A     99.99.1.1
````

Reply:
oh, cute!

Reply:
implemented, really nice :)